### PR TITLE
2838/multiple photos and modal

### PR DESCRIPTION
**Pull request by jaredcwhite**
# Pull Request Template

## Issue Overview

This PR addresses #2838

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

~~Note: needs to be reconciled with #2857 once that's merged into dev.~~ Done.

This PR adds support for multiple images to the ImageCard component, and in addition will display a gallery popup modal when the card is clicked if more than one image is available.

## How Can This Be Tested/Reviewed?

There are a few new stories in Storybook featuring multiple images. https://deploy-preview-2862--dev-storybook-bloom.netlify.app/?path=/story/blocks-image-card--four-images

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [ ] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

**Comment by jaredcwhite** [comment]
While working on this PR, I updated the Modal component to 2nd gen styling. Just now, I noticed that @emilyjablonski's PR https://github.com/bloom-housing/bloom/pull/2857 also does the same! So we'll probably want to review both in tandem and figure out how to reconcile. A quick glance seems to indicate the changeset is pretty similar. 👍

File: ui-components/src/blocks/ImageCard.tsx [hunk 1]
```diff
@@ -59,35 +74,103 @@ const ImageCard = (props: ImageCardProps) => {
...
-        ) : (
-          <div className={"image-card__placeholder"} />
-        )}
-      </figure>
-      {getStatuses()}
-    </div>
+      {props.images && props.images.length > 1 && (
+        <Modal
+          open={showModal}
+          title="Gallery"
```
Comment (emilyjablonski) [inline ui-components/src/blocks/ImageCard.tsx:141]: Should pass in this string so consumers can translate it

**Review by emilyjablonski — CHANGES_REQUESTED** [review]
This is looking so so so good!!
Two things:
* Can we update the test with one that uses the new prop
* Looks like we need a background color on the modal, looking at the story below

Screen Shot 2022-07-14 at 12 23 12 PM

**Comment by jaredcwhite** [comment]
@emilyjablonski @slowbot yeah I was wondering about the background of the modal. Should it fade more to white sort of like how I did it, or fade to gray, or black?

Also, our modal requires a title currently so I just put in "Gallery" but I'm wondering if we need to hide that and just make it an aria label or something, or use some other title?

**Comment by jaredcwhite** [comment]
Updated test added.

**Comment by jaredcwhite** [comment]
@slowbot I've updated the component to accomodate your points there. I do have alt text working for the images in the modal, so I guess we should make sure the CMS can supply that. Otherwise it defaults to the main ImageCard description plus "- photo X"

**Review by emilyjablonski — COMMENTED** [review]
Everything from above LGTM!
One thing I've noticed from the new text is a little big of text overlap on Storybook's small mobile viewport setting, maybe we just need to reduce the size on small screens a little.

Screen Shot 2022-07-20 at 12 34 45 PM

**Review by emilyjablonski — APPROVED** [review]
🥳

**Pull request merged**